src/python_testing/TC_OPCREDS_3_4.py

```diff
@@ -0,0 +1,241 @@
+
+        self.print_step(11, "TH1 sends the UpdateNOC command to the Node Operational Credentials cluster")
+        cmd = opcreds.Commands.UpdateNOC(NOCValue=noc_original, ICACValue=icac_original)
+        resp = await self.send_single_cmd(dev_ctrl=th1_new_fabric_ctrl, node_id=th1_dut_node_id, cmd=cmd)
+        asserts.assert_equal(resp.statusCode, opcreds.Enums.NodeOperationalCertStatusEnum.kInvalidPublicKey,
+                             "Unexpected error code on AddNOC with mismatched CSR")
+
+        self.print_step(12, "TH1 generates a new Trusted Root Certificate and Private Key and saves as new_root_cert and new_root_key so that TH can generate an NOC for UpdateNOC that doesn’t chain to the original root")
+        new_root_cert = new_noc_chain.rcacBytes
+        new_root_key = TLVReader(new_root_cert).get()["Any"][9]
+        print("new_root_key: ", new_root_key)
+
+        self.print_step(13, "TH1 generates a new NOC and ICAC")
+        new_noc_chain = th1_new_fabric_ctrl.IssueNOCChain(csr_update, th1_dut_node_id)
+        noc_update_new_root = csr_update.NOCSRElements
+        icac_update_new_root = new_noc_chain.icacBytes
+
+        self.print_step(14, "TH1 sends the UpdateNOC command to the Node Operational Credentials cluster")
+        cmd = opcreds.Commands.UpdateNOC(NOCValue=noc_update_new_root, ICACValue=icac_update_new_root)
+        resp = await self.send_single_cmd(dev_ctrl=th1_new_fabric_ctrl, node_id=th1_dut_node_id, cmd=cmd)
+        asserts.assert_equal(resp.statusCode, opcreds.Enums.NodeOperationalCertStatusEnum.kInvalidNOC,
+                             "NOCResponse with the StatusCode InvalidNOC")
+
+        self.print_step(15, "TH1 generates a new NOC and ICAC")
+        # new NOC is generated from the NOCSR returned in csr_update with the matter-fabric-id set to a different
+        # value than noc_original. The NOC is signed by new ICA. Save as noc_update_bad_fabric_on_noc.
+        noc_update_bad_fabric_on_noc = csr_update.NOCSRElements
+        # new ICAC is generated with the and matter-fabric-id omitted. ICAC is signed by the original key for
+        # trusted_root_original. Save as icac_update_bad_fabric_on_noc
+        icac_update_bad_fabric_on_noc = new_noc_chain.icacBytes
+
+        self.print_step(16, "TH1 sends the UpdateNOC command to the Node Operational Credentials cluster")
+        cmd = opcreds.Commands.UpdateNOC(NOCValue=noc_update_bad_fabric_on_noc, ICACValue=icac_update_bad_fabric_on_noc)
+        resp = await self.send_single_cmd(dev_ctrl=th1_new_fabric_ctrl, node_id=th1_dut_node_id, cmd=cmd)
+        asserts.assert_equal(resp.statusCode, opcreds.Enums.NodeOperationalCertStatusEnum.kInvalidNOC,
+                             "NOCResponse with the StatusCode InvalidNOC")
+
+        self.print_step(17, "TH1 generates a new NOC and ICAC")
+        noc_update_bad_fabric_on_icac = csr_update.NOCSRElements
```

Comment:
same - this is returning an invalid error for the wrong reasons.

Reply:
Done